src/chain.ts

```diff
@@ -88,6 +88,9 @@ export const chain = (
     client.on('connect', (response, targetSocket, clientHead) => {
         countTargetBytes(sourceSocket, targetSocket);
 
+        // There may be many .on('close') listeners, so we need to increase the limit.
+        targetSocket.setMaxListeners(Infinity);
```

Comment:
Isn't this risky? The warning is there to help detect memory leaks, now we turn this off and if proxy starts crashing, we won't even know why. Maybe increase it too like 1000 instead?